controller_manager:
  ros__parameters:
    update_rate: 30  # Hardware update rate
    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster
    diffbot_base_controller:
      type: diff_drive_controller/DiffDriveController
    gpio_controller:
      type: gpio_controllers/GpioCommandController

diffbot_base_controller:
  ros__parameters:
    left_wheel_names: ["left_wheel_joint"]
    right_wheel_names: ["right_wheel_joint"]
    wheel_separation: 1.016 # approximation. this is equal to base_length.
    wheel_radius: 0.165
    wheel_separation_multiplier: 1.0
    left_wheel_radius_multiplier: 1.0
    right_wheel_radius_multiplier: 1.0
    publish_rate: 30.0  # Tf and odom output rate, bounded by the hardware update_rate above diddy bot
    odom_frame_id: odom
    base_frame_id: base_link
    pose_covariance_diagonal : [0.001, 0.001, 0.001, 0.001, 0.001, 0.01]
    twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.01]
    open_loop: false
    enable_odom_tf: true
    publish_command: true
    cmd_vel_timeout: 0.5
    linear.x.has_velocity_limits: true
    linear.x.has_acceleration_limits: true
    linear.x.max_velocity: 2.0
    linear.x.max_acceleration: 2.0

    angular.z.has_velocity_limits: true
    angular.z.has_acceleration_limits: true
    angular.z.max_velocity: 1.5
    angular.z.max_acceleration: 1.5

gpio_controller:
  ros__parameters:
    type: gpio_controllers/GpioCommandController
    gpios:
      - imprimis_data
    command_interfaces:
      imprimis_data:
        - interfaces:
          - dummy_gpio_cmd
    state_interfaces:
      imprimis_data:
        - interfaces:
          - imuHeading
          - manualMode
          - boardBConnected
          - gpsFix
          - gpsLat
          - gpsLong
          - gpsAlt
          - gpsHdop